### ISSUE TITLE
Folders take forever to open

### ISSUE THREAD

**Issue by UltimateByte** [lrsjng/h5ai#458]
Hi there,

I just installed this great file explorer, and many thanks for it !
I'm just encountering 2 issues, and especially one that i would like to get rid of so i can really be happy with it.

- The first one : i just posted into the original issue : videos don't stop loading after closing them (saw that looking my network usage)

- The second one is way more important : folders that takes forever to open up, and by "forever" i mean about 30 seconds.

May i mention my server is an OVH SP-128 (Hexacore, 128GB RAM) that isn't nearly over-charged (more than 90% cpu free, more than 90% RAM free...) as i'm writing this.
I saw processes using 100% of a core and lagging, but i've never seen the server lag yet after one month and a half, so server performance could not be the reason. (unless you think 48 days uptime could be the reason... but i don't think so).
It's running Debian 8 x64 with all updates, latest Apache2 and PHP5.
I'm running several wordpress websites without any lag in any sort.
For this file browser, without the h5ai browser, i can navigate through folders without any delay.
When the issue occurs, i can see a www-data process using more than 22% of a thread.

I'm suspecting that i messed around with my .htaccess, or that files, belonging to debian-transmission, (those are files from a seedbox) may cause the issue.
Right now, i only got 1 .htaccess at the root of my website and no .htaccess file into the _h5ai folder. I'm not sure of the right way, so i tried duplicating the .htaccess into the _h5ai folder but it doesn't change anything. I tried several restarts of apache 2 too.

I can provide any information to try diagnose this issue, just... as it's a seedbox, i'd prefer not posting the link right here.

Thanks for your attention.

**Comment by UltimateByte** [comment]
Well, after some more investigation, i'm watched my logs (bad habbit not doing it earlier). It's spamming this : 

[Mon Jun 15 04:55:05.381614 2015] [:error] [pid 9894] [client xx.xx.xx.xx:58363] PHP Warning:  file_put_contents(/home/seedbox/downloads/_h5ai/cache/cmds.json): failed to open stream: Permission denied in /home/seedbox/downloads/_h5ai/server/php/inc/class-util.php on line 81, referer: http://seedbox.xxx.xx/

Let's play with permissions then.

**Comment by UltimateByte** [comment]
I really need this up and running guyz, any help would be appreciated !

I when through more investigation : 

With logs set to debug, i've got a slow spamming of this in error.log : 
"[authz_core:debug] [pid 7981] mod_authz_core.c(835):  [client 89.2.xx.xx:53787] AH01628: authorization result: granted (no directives), referer: http://..."

The same thing with several PID.

**Comment by UltimateByte** [comment]
I went though all apache2 and php5 configuration, googled for hours, i don't have a single clue...

**Comment by lrsjng** [comment]
@UltimateByte hey, sounds strange, do you have a live link for me to have a look?

**Comment by UltimateByte** [comment]
I'm glad you're here ! I'm pretty sure you're the only one that can get this solved.
So i'm now running the latest dev build with the exact same issue.
http://seedbox.mesnie.org/
See some websites onto the same server : 
http://www.mesnie.org/
http://www.lrob.fr/
Empty websites (folder browsing) are smooth too : 
http://darkrp.mesnie.org/

I'll try what you said here : https://github.com/lrsjng/h5ai/issues/459
Change   
if (PHP_INT_SIZE < 8) { 
to 
if (false) 

And edit this post to tell you what it does

**Comment by Sakarias-Longhorn** [comment]
Yep, I can confirm, some folders takes forever to load, I tried to create empty test folder, and they're slow to load sometimes, I even tried to create fake files inside them with "fallocate" and they're still slow to load. If I disable h5ai by renaming it's folder, they load instantly with the default apache listing.

**Comment by Dragnell87** [comment]
Any update on this? at the moment increasing max_execution_time  to 300, same for 
fastcgi_connect_timeout does show the folder content however the site is unresponsive while doing the task.

**Comment by n0rc** [comment]
I had a similar problem where browsing through my folders with h5ai took a long time to load specific directories.

I could solve it by checking `_h5ai/public/index.php` and making sure that *all* requirements for optional features are met (all green). I had to make some changes to my PHP configuration but since then everything works flawlessly.

**Comment by Dragnell87** [comment]
@n0rc Would you mind to explain what changes you did, thanks to you reply i was able to put almost all requirement in green one exception Private Cache directory,

**Comment by n0rc** [comment]
@Dragnell87 I set the permissions for both caching folders appropriately and removed the relevant functions from `disabled_functions` in my php.ini (see 5th bullet point of the "Hints" section at https://larsjung.de/h5ai/).

**Comment by Dragnell87** [comment]
@n0rc Thank, it is all working now.
